extension/src/popup/views/GrantAccess.tsx

```diff
@@ -43,9 +46,9 @@ const RejectButtonEl = styled(Button)`
 export const GrantAccess = () => {
   const location = useLocation();
   const dispatch = useDispatch();
-  const decodedTab = atob(location.search.replace("?", ""));
-  const tabToUnlock = decodedTab ? JSON.parse(decodedTab) : {};
-  const { url, title } = tabToUnlock;
+  const { url } = parsedSearchParam(location.search);
+  const u = new URL(url);
+  const punycodedDomain = punycode.toASCII(u.hostname);
```

Comment:
hm, we have to convert to `new URL` and then grab this `hostname` a few times. Maybe we should create a helper that accepts a url string and returns the hostname

Reply:
Updated!